src/views/SystemProfile/index.tsx

```diff
@@ -52,6 +71,135 @@ import SystemSubNav from './components/SystemSubNav/index';
 
 import './index.scss';
 
+export function formatDate(v: string) {
+  return DateTime.fromISO(v).toLocaleString(DateTime.DATE_FULL);
+}
+
+/**
+ * Get the ATO Status from certain date properties and flags.
+ */
+function getAtoStatus(
+  // eslint-disable-next-line camelcase
+  cedarAuthorityToOperate?: GetSystemProfile_cedarAuthorityToOperate
+): AtoStatus {
+  // `ato.dateAuthorizationMemoExpires` will be null if tlcPhase is Initiate|Develop
+
+  // No ato if it doesn't exist
+  if (!cedarAuthorityToOperate) return 'No ATO';
+
+  // return 'In Progress'; // tbd
+
+  const expiry = cedarAuthorityToOperate.dateAuthorizationMemoExpires;
+  if (!expiry) return 'No ATO';
+
+  const date = new Date().toISOString();
+  // console.log(date, expiry);
+
+  if (date >= expiry) return 'Expired';
+  if (date >= subDays(expiry, ATO_STATUS_DUE_SOON_DAYS).toISOString())
+    return 'Due Soon';
+  return 'Active';
+}
+
+/**
+ * Get Development Tags which are derived from various other properties.
+ */
+export function getDevelopmentTags(
+  // eslint-disable-next-line camelcase
+  cedarSystemDetails: GetSystemProfile_cedarSystemDetails
+): DevelopmentTag[] {
+  const tags: DevelopmentTag[] = [];
+  if (cedarSystemDetails.systemMaintainerInformation.agileUsed === true) {
+    tags.push('Agile Methodology');
+  }
+  return tags;
+}
+
+/**
+ * Get a list of UrlLocations found from Cedar system Urls and Deployments.
+ * A `UrlLocation` is extended from a Cedar Url with some additional parsing
+ * and Deployment assignments.
+ */
+function getLocations(
+  // eslint-disable-next-line camelcase
+  cedarSystemDetails: GetSystemProfile_cedarSystemDetails
+): UrlLocation[] {
+  return cedarSystemDetails.urls.map(url => {
+    // Find a deployment from matching its type with the url host env
+    const hostenv = url.urlHostingEnv;
+    const deployment = cedarSystemDetails.deployments.filter(
```

Comment:
Two things here:
1. There's a small potential logic bug, though I don't know if it would ever occur in practice. If `hostenv` is null and there's a deployment with `deploymentType: null`, they'll match up, which might mess up finding the provider later. You could prevent that with adding `&& hostenv !== null` or something to the predicate for finding `deployment`.
2. It might be easier to use `find()` instead of `filter()`, since you're just using `deployment[0]` anyways.

Reply:
Followup PR for this: https://github.com/CMSgov/easi-app/pull/1674